### ISSUE TITLE
NoneType is not callable when ctrl-c twice for kill

### ISSUE THREAD

**Issue by samphippen** [docker/compose#620]
```
^CGracefully stopping... (press Ctrl+C again to force)
Stopping kya_workercorpus_1...
^CKilling kya_workercorpus_1...
Killing kya_wordpress_1...
Killing kya_mysqldb_1...
Killing kya_redis_1...
Killing kya_db_1...
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/Users/ben/fig/build/fig/out00-PYZ.pyz/threading", line 810, in __bootstrap_inner
  File "/Users/ben/fig/build/fig/out00-PYZ.pyz/threading", line 763, in run
  File "/Users/ben/fig/build/fig/out00-PYZ.pyz/fig.cli.multiplexer", line 41, in _enqueue_output
  File "/Users/ben/fig/build/fig/out00-PYZ.pyz/fig.cli.log_printer", line 56, in _make_log_generator
  File "/Users/ben/fig/build/fig/out00-PYZ.pyz/fig.container", line 127, in wait
  File "/Users/ben/fig/build/fig/out00-PYZ.pyz/fig.packages.docker.client", line 764, in wait
  File "/Users/ben/fig/build/fig/out00-PYZ.pyz/fig.packages.docker.client", line 65, in _post
  File "/Users/ben/fig/build/fig/out00-PYZ.pyz/requests.sessions", line 425, in post
  File "/Users/ben/fig/build/fig/out00-PYZ.pyz/requests.sessions", line 383, in request
  File "/Users/ben/fig/build/fig/out00-PYZ.pyz/requests.sessions", line 486, in send
  File "/Users/ben/fig/build/fig/out00-PYZ.pyz/requests.adapters", line 391, in send
  File "/Users/ben/fig/build/fig/out00-PYZ.pyz/requests.adapters", line 167, in build_response
  File "/Users/ben/fig/build/fig/out00-PYZ.pyz/requests.models", line 526, in __init__
  File "/Users/ben/fig/build/fig/out00-PYZ.pyz/requests.structures", line 69, in __init__
  File "/Users/ben/fig/build/fig/out00-PYZ.pyz/_abcoll", line 540, in update
  File "/Users/ben/fig/build/fig/out00-PYZ.pyz/abc", line 144, in __instancecheck__
  File "/Users/ben/fig/build/fig/out00-PYZ.pyz/abc", line 174, in __subclasscheck__
  File "/Users/ben/fig/build/fig/out00-PYZ.pyz/_weakrefset", line 59, in __iter__
TypeError: 'NoneType' object is not callable
```

got this error when trying to shut down my project.

**Comment by jamshid** [comment]
Yes, in my case, I had an empty environment in `fig.yml`. Feeding google with the new docker-compose stack trace.
```
  environment:
    - 
```
```
$ docker-compose up -d
Creating foo_test_1...
Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "/Users/ben/fig/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 31, in main
  File "/Users/ben/fig/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 21, in sys_dispatch
  File "/Users/ben/fig/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 27, in dispatch
  File "/Users/ben/fig/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 24, in dispatch
  File "/Users/ben/fig/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 59, in perform_command
  File "/Users/ben/fig/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 445, in up
  File "/Users/ben/fig/build/docker-compose/out00-PYZ.pyz/compose.project", line 184, in up
  File "/Users/ben/fig/build/docker-compose/out00-PYZ.pyz/compose.service", line 259, in recreate_containers
  File "/Users/ben/fig/build/docker-compose/out00-PYZ.pyz/compose.service", line 227, in create_container
  File "/Users/ben/fig/build/docker-compose/out00-PYZ.pyz/compose.service", line 453, in _get_container_create_options
  File "/Users/ben/fig/build/docker-compose/out00-PYZ.pyz/compose.service", line 640, in merge_environment
  File "/Users/ben/fig/build/docker-compose/out00-PYZ.pyz/compose.service", line 640, in <genexpr>
  File "/Users/ben/fig/build/docker-compose/out00-PYZ.pyz/compose.service", line 648, in split_env
TypeError: argument of type 'NoneType' is not iterable
```

**Comment by bastibense** [comment]
I think I found a similar issue with docker-compose. Let me know if not directly related and I'll open a separate issue.

    $ docker-compose up
    Traceback (most recent call last):
      File "/usr/local/Cellar/fig/1.2.0/libexec/bin/docker-compose", line 9, in <module>
        load_entry_point('docker-compose==1.2.0', 'console_scripts', 'docker-compose')()
      File "/usr/local/Cellar/fig/1.2.0/libexec/lib/python2.7/site-packages/compose/cli/main.py", line 31, in main
        command.sys_dispatch()
      File "/usr/local/Cellar/fig/1.2.0/libexec/lib/python2.7/site-packages/compose/cli/docopt_command.py", line 21, in sys_dispatch
        self.dispatch(sys.argv[1:], None)
      File "/usr/local/Cellar/fig/1.2.0/libexec/lib/python2.7/site-packages/compose/cli/command.py", line 27, in dispatch
        super(Command, self).dispatch(*args, **kwargs)
      File "/usr/local/Cellar/fig/1.2.0/libexec/lib/python2.7/site-packages/compose/cli/docopt_command.py", line 24, in dispatch
        self.perform_command(*self.parse(argv, global_options))
      File "/usr/local/Cellar/fig/1.2.0/libexec/lib/python2.7/site-packages/compose/cli/command.py", line 57, in perform_command
        verbose=options.get('--verbose'))
      File "/usr/local/Cellar/fig/1.2.0/libexec/lib/python2.7/site-packages/compose/cli/command.py", line 76, in get_project
        config.load(config_path),
      File "/usr/local/Cellar/fig/1.2.0/libexec/lib/python2.7/site-packages/compose/config.py", line 55, in load
        return from_dictionary(load_yaml(filename), working_dir=working_dir, filename=filename)
      File "/usr/local/Cellar/fig/1.2.0/libexec/lib/python2.7/site-packages/compose/config.py", line 65, in from_dictionary
        service_dict = loader.make_service_dict(service_name, service_dict)
      File "/usr/local/Cellar/fig/1.2.0/libexec/lib/python2.7/site-packages/compose/config.py", line 89, in make_service_dict
        return process_container_options(service_dict, working_dir=self.working_dir)
      File "/usr/local/Cellar/fig/1.2.0/libexec/lib/python2.7/site-packages/compose/config.py", line 175, in process_container_options
        service_dict['build'] = resolve_build_path(service_dict['build'], working_dir=working_dir)
      File "/usr/local/Cellar/fig/1.2.0/libexec/lib/python2.7/site-packages/compose/config.py", line 342, in resolve_build_path
        _path = expand_path(working_dir, build_path)
      File "/usr/local/Cellar/fig/1.2.0/libexec/lib/python2.7/site-packages/compose/config.py", line 383, in expand_path
        return os.path.abspath(os.path.join(working_dir, path))
      File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/posixpath.py", line 75, in join
        if b.startswith('/'):
    AttributeError: 'NoneType' object has no attribute 'startswith'

and my `docker-compose.yml` looked like this:


    db:
      image: postgres

    portal:
      build:
      links:
        - db
      ports:
        - "8000:8000"
      volumes:
        - .:/


It seems to work fine when I add a period (`.`) after `build:` like this:


    db:
      image: postgres

    portal:
      build: .
      links:
        - db
      ports:
        - "8000:8000"
      volumes:
        - .:/

**Comment by li0nel** [comment]
Got the same problem, reinstalled compose and boot2docker, but it was just a malformed docker-compose.yml...

**Comment by dnephin** [comment]
There area few different issues being reported here.

The config validation issues should be resolved by #129 in the 1.5.0 release.
The original issue was most likely a bug in requests, which we've upgraded a couple times since this issue was created.
